pkg/controller/status.go

```diff
@@ -56,9 +56,15 @@ func (c *Controller) handleIngressStatus(ingress *networkingv1.Ingress, vs *isti
 
 // getGatewaysForVirtualService will get the gateways associated with the Virtual Service.
 func (c *Controller) getGatewaysForVirtualService(vs *istionetworkingv1beta1.VirtualService) ([]*istionetworkingv1beta1.Gateway, error) {
+	return c.getGatewaysByName(vs.Spec.Gateways, vs.Namespace)
```

Comment:
Not sure why I did this now... It's actually kind of useless. The VS won't be `nil` when we're using it since the struct will still be populated with the values we need.